libs/api/src/lib/memberContext.ts

```diff
@@ -622,17 +622,38 @@ export class MemberContext implements MemberContext {
   }
 
   async createSubscription(
-    subscriptionClient: PrismaClient['subscription'],
+    prisma: PrismaClient,
     userID: string,
     paymentMethodId: string,
     paymentPeriodicity: PaymentPeriodicity,
     monthlyAmount: number,
     memberPlanId: string,
     properties: Pick<MetadataProperty, 'key' | 'value' | 'public'>[],
     autoRenew: boolean,
+    extendable: boolean,
     startsAt?: Date | string
   ): Promise<{subscription: SubscriptionWithRelations; invoice: InvoiceWithItems}> {
-    const subscription = await subscriptionClient.create({
+    if (!extendable && autoRenew) {
+      throw new Error("You can't create a non extendable subscription that is autoRenew!")
+    }
+
+    const memberPlan = await prisma.memberPlan.findUnique({where: {id: memberPlanId}})
+    const memberPlanSubscriptionCount = await prisma.subscription.count({
+      where: {
+        userID,
+        memberPlanID: memberPlanId
+      }
+    })
+
+    if (memberPlan.maxCount && memberPlan.maxCount <= memberPlanSubscriptionCount) {
```

Comment:
In case maxCount === 0, this error would never hit. From a users perspective, this would probably not be intended.
I would test this functionality with unit tests using undefined, null, 0 and 1 as input values.